fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/rdf/impl/TypeRdfContext.java

```diff
@@ -58,71 +61,36 @@ public TypeRdfContext(final FedoraResource resource,
 
         //include rdf:type for primaryType, mixins, and their supertypes
         concatRdfTypes();
+        if (resource() instanceof NonRdfSource) {
+            // gather versionability info from the parent
+            if (resource().getNode().getParent().isNodeType("mix:versionable")) {
```

Comment:
The `NonRdfSourceDescription` is the parent of the `NonRdfSource`. When it is versionable, so is the `NonRdfSource`. In fact, the tests wouldn't pass if this weren't so (see `FedoraVersionsIT.testDatastreamAutoMixinAndRevert()`). I still don't understand what you claim has been lost here.